File: src/com/owlrobotics/pedropathingvisualizer/pedropathing/util/PlaneOrigin.java
package com.owlrobotics.pedropathingvisualizer.pedropathing.util;

import java.awt.*;

public enum PlaneOrigin {
    CENTER,
    TOP_LEFT,
    TOP_RIGHT,
    BOTTOM_LEFT,
    BOTTOM_RIGHT;

    public Dimension getOrigin(int windowSize) {

        return switch (this) {
            case CENTER -> new Dimension(windowSize / 2, windowSize / 2);
            case TOP_LEFT -> new Dimension(0, 0);
            case TOP_RIGHT -> new Dimension(windowSize, 0);
            case BOTTOM_LEFT -> new Dimension(0, windowSize);
            case BOTTOM_RIGHT -> new Dimension(windowSize, windowSize);
        };
    }
}

File: src/com/owlrobotics/pedropathingvisualizer/pedropathing/util/PlaneOrigin.kt
package com.owlrobotics.pedropathingvisualizer.pedropathing.util

import java.awt.Dimension

enum class PlaneOrigin {
    CENTER,
    TOP_LEFT,
    TOP_RIGHT,
    BOTTOM_LEFT,
    BOTTOM_RIGHT;

    fun getOrigin(windowSize: Int): Dimension {
        return when (this) {
            CENTER -> Dimension(windowSize / 2, windowSize / 2)
            TOP_LEFT -> Dimension(0, 0)
            TOP_RIGHT -> Dimension(windowSize, 0)
            BOTTOM_LEFT -> Dimension(0, windowSize)
            BOTTOM_RIGHT -> Dimension(windowSize, windowSize)
        }
    }
}
